testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveViewsCompatibility.java

```diff
@@ -0,0 +1,133 @@
+                    .hasMessageContaining("Not an Iceberg table: default.hive_table");
+            onTrino().executeQuery("CREATE VIEW iceberg.default.iceberg_view_qualified_iceberg AS SELECT * FROM iceberg.default.iceberg_table");
+            onTrino().executeQuery("CREATE VIEW iceberg.default.iceberg_view_unqualified_iceberg AS SELECT * FROM iceberg_table");
+
+            // both hive and iceberg catalogs should list all the views.
+            assertThat(onTrino().executeQuery("SHOW TABLES FROM hive.default"))
+                    .containsOnly(ImmutableList.<QueryAssert.Row>builder()
+                            .addAll(hivePreexistentTables)
+                            .add(row("hive_table"))
+                            .add(row("iceberg_table")) // TODO; should this be filtered out?
+                            .add(row("hive_view_qualified_hive"))
+                            .add(row("hive_view_unqualified_hive"))
+                            .add(row("hive_view_qualified_iceberg"))
+                            .add(row("hive_view_unqualified_iceberg"))
+                            .add(row("iceberg_view_qualified_hive"))
+                            .add(row("iceberg_view_qualified_iceberg"))
+                            .add(row("iceberg_view_unqualified_iceberg"))
+                            .build());
+
+            assertThat(onTrino().executeQuery("SHOW TABLES FROM iceberg.default"))
+                    .containsOnly(ImmutableList.<QueryAssert.Row>builder()
+                            .addAll(icebergPreexistentTables)
+                            .add(row("hive_table"))
+                            .add(row("hive_view_qualified_hive"))
+                            .add(row("hive_view_unqualified_hive"))
+                            .add(row("hive_view_qualified_iceberg"))
+                            .add(row("hive_view_unqualified_iceberg"))
+                            .add(row("iceberg_view_qualified_hive"))
+                            .add(row("iceberg_view_qualified_iceberg"))
+                            .add(row("iceberg_view_unqualified_iceberg"))
+                            .build());
+
+            // try to access all views via hive catalog
+            assertThat(onTrino().executeQuery("SELECT * FROM hive.default.hive_view_qualified_hive")).containsOnly(row(1));
+            assertThat(onTrino().executeQuery("SELECT * FROM hive.default.hive_view_unqualified_hive")).containsOnly(row(1));
+            assertThat(onTrino().executeQuery("SELECT * FROM hive.default.hive_view_qualified_iceberg")).containsOnly(row(2));
+            assertThatThrownBy(() -> onTrino().executeQuery("SELECT * FROM hive.default.hive_view_unqualified_iceberg"))
+                    // hive connector tries to read from iceberg table
+                    // TODO; make query fail with nicer message
+                    .hasMessageContaining("Unable to create input format org.apache.hadoop.mapred.FileInputFormat");
```

Comment:
Should we have a test that this works when redirects https://github.com/trinodb/trino/pull/8340 are possible?
cc @phd3 

(no change requested here)

Reply:
makes sense